### PR TITLE
convert to svg-bytes instead of png-bytes; auto-scaling large images

### PR DESCRIPTION
**Pull request by lihebi**
Issue #469

### CODE REVIEW

File: racket-repl.el [hunk 1]
```diff
@@ -700,7 +700,13 @@ A value for the variable `comint-output-filter-functions'."
           (delete-region begin end)
           (goto-char begin)
           (if (and racket-images-inline (display-images-p))
-              (insert-image (create-image file) "[image]")
+              (insert-image (if (> racket-images-max-width 0)
```
Comment (greghendershott) [inline racket-repl.el:703]: Should we also check `(image-type-available-p 'imagemagick)` here?

File: racket-custom.el [hunk 2]
```diff
@@ -243,6 +243,13 @@ Defaults to a regexp ignoring all inputs of 0, 1, or 2 letters."
   :safe #'booleanp
   :group 'racket-repl)
 
+(defcustom racket-images-max-width -1
```
Comment (greghendershott) [inline racket-custom.el:246]: Maybe instead default to `0`?

File: racket-custom.el [hunk 3]
```diff
@@ -243,6 +243,13 @@ Defaults to a regexp ignoring all inputs of 0, 1, or 2 letters."
...
+  "The max width of inlined images. Negative values disable auto-scaling."
```
Comment (greghendershott) [inline racket-custom.el:247]: Maybe add a second line, something like "Note: This option only works when Emacs is built with ImageMagick (libMagickWand) support." I'm just using the phrase from the Emacs Lisp Info manual.

**Comment by greghendershott** [comment]
Thanks! This looks great. I had just a few small questions, inline comments above.

**Comment by greghendershott** [comment]
Thanks for making those changes!

I pushed a few more commits.

What do you think of those?

Do they work for you -- if you set the customization now named `racket-imagemagick-props` to the value `'(:max-width 1000)`)?

**Comment by lihebi** [comment]
Thanks. Just tested, that works great for me.

**Comment by greghendershott** [comment]
Great. I squashed down to one commit. I'll merge to `master` after I get all green from Travis CI.

**Comment by greghendershott** [comment]
@timjervis Oh. Sorry! You shouldn't need to change your Racket programs. I'll open an issue for this, and `@` you and @lihebi.

**Comment by timjervis** [comment]
Thanks!

**Pull request merged**